### PR TITLE
短链接 &&  重置b站短链接与bv链接到av，替换 twitter.com 到 fxtwitter.com

### PR DESCRIPTION
**Pull request by Anthony-Hoo**
### Summary
🔗🎙️🔍

This pull request adds two new features to the bot: url filter and slink. The url filter feature can filter and convert URLs in the messages, such as transforming bilibili BV numbers to AV numbers, removing tracking parameters from twitter URLs, and using a custom domain for shortening URLs. The slink feature can create short URLs using a GitHub repository as a backend. The pull request also adds new configuration options, dependencies, middleware, handler, and commands for these features, and updates the README.md file accordingly.

> _We are the masters of the link_
> _We filter, shorten, and convert_
> _We use the power of the `github` and `content_filter`_
> _We are the bot of the slink_

### Walkthrough
*  Add new files for the url filter and slink features ([link](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-96bf86aff0f263031523cef568901111ca5bcf38e5a148b47b970d7a5113505fR1-R158), [link](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-4df197749bb40eefb1da06cc53a262c7e07fd00652acc832cbfce745f39d6227R1-R132), [link](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-3cd3b156fcd446b3f4cd3aad6921491e8680d8379338ba5fe841aa6b82029282R1-R38), [link](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-9d341698660ecd4483dea4c18980a12532b41bf1af84a896d9c934531f76f81eR1-R30))
*  Add new dependencies for the url filter and slink features ([link](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R6-R7), [link](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6L16-R21), [link](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R36), [link](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R50), [link](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R62-R64), [link](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R75))
*  Modify the `config.yaml` file to add new sections for the github and content_filter settings ([link](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-d8d0422389f03d783e32e627250fe29834bd09c6361640d1ff00661dd6820034L71-R89))
*  Modify the `config` package to add new structs and methods for the github and content_filter settings ([link](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-fe44f09c4d5977b5f5eaea29170b6a0748819c9d02271746a20d81a5f3efca17R20), [link](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-fe44f09c4d5977b5f5eaea29170b6a0748819c9d02271746a20d81a5f3efca17L40-R53), [link](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-fe44f09c4d5977b5f5eaea29170b6a0748819c9d02271746a20d81a5f3efca17L69-R84), [link](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-fe44f09c4d5977b5f5eaea29170b6a0748819c9d02271746a20d81a5f3efca17R128-R129), [link](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-fe44f09c4d5977b5f5eaea29170b6a0748819c9d02271746a20d81a5f3efca17R158-R159))
*  Modify the `main` package to add a new middleware and a new handler for the url filter and slink features ([link](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261L93-R93), [link](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R145-R146), [link](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R326-R339))
*  Add new commands to the `README.md` file to document the new features and provide usage examples ([link](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R82-R85))

### CODE REVIEW

**Review by Rachel030219 — APPROVED** [review]

**Review by icceey — CHANGES_REQUESTED** [review]
- 希望域名可以配置喵 [在这里](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-4df197749bb40eefb1da06cc53a262c7e07fd00652acc832cbfce745f39d6227R66)
- 希望可以移除这个id喵 [在这里](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-d8d0422389f03d783e32e627250fe29834bd09c6361640d1ff00661dd6820034R86-R89)

File: go.mod [hunk 1]
```diff
@@ -3,6 +3,8 @@ module csust-got
 go 1.20
 
 require (
+	github.com/google/go-github/v35 v35.3.0
```
Comment (hugefiver) [inline go.mod:6]: 这是啥

File: go.mod [hunk 2]
```diff
@@ -13,8 +15,10 @@ require (
 	github.com/stretchr/testify v1.8.2
 	go.uber.org/zap v1.24.0
 	golang.org/x/net v0.9.0
+	golang.org/x/oauth2 v0.5.0
 	golang.org/x/time v0.3.0
 	gopkg.in/telebot.v3 v3.1.3
+	mvdan.cc/xurls/v2 v2.5.0
```
Comment (hugefiver) [inline go.mod:21]: 这又是啥

File: base/short_url.go [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+	source := rand.NewSource(time.Now().UnixNano())
+	randomGen := rand.New(source)
+	for i := 0; i < 6; i++ {
+		identifier.WriteByte(identifierChars[randomGen.Intn(len(identifierChars))])
+	}
+	return identifier.String()
+}
+
+// 将URL添加到CSV文件并提交到GitHub仓库
+func addURLToCSV(client *gh.Client, owner, repo, path, url string) (string, error) {
```
Comment (hugefiver) [inline base/short_url.go:31]: 原子性吗

File: base/short_url.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+	"go.uber.org/zap"
+	"golang.org/x/oauth2"
+	. "gopkg.in/telebot.v3"
+	"math/rand"
+	"strings"
+	"time"
+)
+
+// 生成一个6位的随机标识符
+func generateIdentifier() string {
```
Comment (hugefiver) [inline base/short_url.go:19]: https://github.com/CSUSTers/csust-got/blob/master/util/randstr.go#L38

File: base/short_url.go (hunk 3 above)
Comment (Anthony-Hoo) [reply base/short_url.go:31]: 原

File: base/short_url.go (hunk 4 above)
Comment (Anthony-Hoo) [reply base/short_url.go]: > https://github.com/CSUSTers/csust-got/blob/master/util/randstr.go#L38

改了 [这里](https://github.com/CSUSTers/csust-got/pull/202/commits/94085abc5f17c36dc85602cc835c44d08a114d56#diff-4df197749bb40eefb1da06cc53a262c7e07fd00652acc832cbfce745f39d6227R47)

File: base/short_url.go (hunk 3 above)
Comment (Anthony-Hoo) [reply base/short_url.go:18]: 感觉不是很需要关心这个，因为如果别人中途修改了就提交不上去了

File: go.mod (hunk 2 above)
Comment (Anthony-Hoo) [reply go.mod:21]: 那咋办嘛，你写一个url处理库吗

**Comment by Anthony-Hoo** [comment]
> * 希望域名可以配置喵 [在这里](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-4df197749bb40eefb1da06cc53a262c7e07fd00652acc832cbfce745f39d6227R66)
> * 希望可以移除这个id喵 [在这里](https://github.com/CSUSTers/csust-got/pull/202/files?diff=unified&w=0#diff-d8d0422389f03d783e32e627250fe29834bd09c6361640d1ff00661dd6820034R86-R89)

都改了

**Review by icceey — APPROVED** [review]
Approved

File: base/short_url.go (hunk 3 above)
Comment (hugefiver) [reply base/short_url.go:18]: 用api的话会直接覆盖的

Comment (Anthony-Hoo) [reply base/short_url.go:18]: 覆盖了问题也不大，page里面还有就行了

**Review by hugefiver — APPROVED** [review]

**Review by Rachel030219 — APPROVED** [review]

**Pull request merged**